internal/controller/lcore/assets.go

```diff
@@ -0,0 +1,1206 @@
+package lcore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/openshift/lightspeed-operator/internal/controller/reconciler"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	"github.com/openshift/lightspeed-operator/internal/controller/utils"
+)
+
+// Service account for running LCore server
+func GenerateServiceAccount(r reconciler.Reconciler, cr *olsv1alpha1.OLSConfig) (*corev1.ServiceAccount, error) {
+	sa := corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      utils.OLSAppServerServiceAccountName,
+			Namespace: r.GetNamespace(),
+		},
+	}
+
+	if err := controllerutil.SetControllerReference(cr, &sa, r.GetScheme()); err != nil {
+		return nil, err
+	}
+
+	return &sa, nil
+}
+
+// SAR = SubjectAccessReview
+// SARClusterRole provides permissions for the OLS Application Server to perform
+// authentication and authorization checks for users accessing the OLS API.
+func GenerateSARClusterRole(r reconciler.Reconciler, cr *olsv1alpha1.OLSConfig) (*rbacv1.ClusterRole, error) {
+	role := rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: utils.OLSAppServerSARRoleName,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{"authorization.k8s.io"},
+				Resources: []string{"subjectaccessreviews"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups: []string{"authentication.k8s.io"},
+				Resources: []string{"tokenreviews"},
+				Verbs:     []string{"create"},
+			},
+			{
+				APIGroups: []string{"config.openshift.io"},
+				Resources: []string{"clusterversions"},
+				Verbs:     []string{"get", "list"},
+			},
+			{
+				APIGroups:     []string{""},
+				Resources:     []string{"secrets"},
+				ResourceNames: []string{"pull-secret"},
+				Verbs:         []string{"get"},
+			},
+			{
+				NonResourceURLs: []string{"/ols-access"},
```

Comment:
lightspeed stack requires access to `ls-access` , no need for `ols-access`.
this is for accessing the liveness/readiness endpoint only